zCore/src/main.rs

```diff
@@ -121,7 +121,8 @@ pub extern "C" fn rust_main(hartid: usize, device_tree_paddr: usize) -> ! {
     });
 
     // 正常由bootloader载入文件系统镜像到内存, 这里不用，而使用后面的virtio
-    let dummy = unsafe { core::slice::from_raw_parts_mut(0 as *mut u8, 0) };
+    // 使用Box创建一个dummy来绕过编译器的参数检查
+    let dummy = Box::leak(Box::new([]));
     main(dummy, boot_info.cmdline);
```

Comment:
直接使用空 slice `&[]` 即可
```suggestion
    main(&[], boot_info.cmdline);
```